synapse/util/caches/treecache.py

```diff
@@ -35,17 +35,17 @@ class TreeCache:
         root = {key_1: {key_2: _value}}
     """
 
-    def __init__(self):
-        self.size = 0
+    def __init__(self) -> None:
+        self.size: int = 0
         self.root = TreeCacheNode()
 
-    def __setitem__(self, key, value):
-        return self.set(key, value)
+    def __setitem__(self, key, value) -> None:
```

Comment:
It'd be nice if we could have typed this while we're here, but this is already a *huge* PR.

Reply:
will come back in another PR to pick this up :)

Reply:
will come back in another PR to pick this one up :)